fix: _consume_uphold_transaction

gahrb opened the pull request:
Closes #(issue_number)

## Checklist

- [ ] The PR modified the frontend, and updated the [user guide](https://github.com/rotki/rotki/blob/develop/docs/usage_guide.rst) to reflect the changes.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/rotki/rotki?pullRequest=4839) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/rotki/rotki?pullRequest=4839) before we can accept your contribution.<br/><hr/>**gahrb** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/rotki/rotki?pullRequest=4839) it.</sub>

yabirgb commented:
Hello @gahrb ! We have a CLA that you would need to sign before contributing to the repo. Can you provide an explanation to the PR?

LefterisJP reviewed (commented):
Hey @gahrb!

Thanks for the PR. Can you maybe explain what it is you are doing and why?

Also some style guides: We use `''` everywhere and all function definitions that split their arguments in multiple lines need 2 indentation levels

gahrb commented:
Thanks @LefterisJP  and @yabirgb for getting back this quickly. I'm without my laptop until Tuesday and will get back to you by then. Marking it as draft.

gahrb commented:
Hi @yabirgb & @LefterisJP,

I changed the commit to match your style guides and signed the CLA.

About the pull request:
- I found the data_import modules to be underrepresented in the tests
- At the same time they all seem to be doing the same: Take some exchange's export, bring it into rotki's data format and import the data

Since I haven't contributed yet to rotki, I found this a good and easy point of entrance. I picked uphold as an example and added the following changes:

- Create helper functions that can be used by each exchange's import module
- Call the corresponding functions (as a first step only) by uphold.

Let me know if this is of help and if the changes lgty. I would then mark the PR as ready.

nebolax commented on Makefile:
```diff
@@ -38,3 +38,4 @@ test-assets:
 	python pytestgeventwrapper.py rotkehlchen/tests/exchanges/test_independentreserve.py::test_assets_are_known
 	python pytestgeventwrapper.py rotkehlchen/tests/unit/test_zerionsdk.py::test_protocol_names_are_known
 	python pytestgeventwrapper.py rotkehlchen/tests/unit/test_zerionsdk.py::test_query_all_protocol_balances_for_account
+	python pytestgeventwrapper.py rotkehlchen/tests/data_import/
```
I don't think that this change needed, since as you can see from the command name `test-assets`, it is supposed to test assets, not data import

nebolax commented on rotkehlchen/data_import/importers/_import_helpers.py:
```diff
@@ -0,0 +1,241 @@
+    return origin == destination
+
+
+def is_same_asset(origin_asset: Asset, destination_asset: Asset) -> bool:
```
What is the idea behind this function? I would say that is cleaner to use `asset1 == asset2` directly

nebolax commented on rotkehlchen/data_import/importers/_import_helpers.py:
```diff
@@ -0,0 +1,241 @@
+) -> bool:
+    """Check if the transaction is a withdrawal."""
+    if not is_same_asset(origin_asset, destination_asset):
+        raise ValueError(
```
We try to avoid raising `ValueError` as much as possible and use more specific errors instead. For example, we have `InputError`

nebolax commented on rotkehlchen/data_import/importers/_import_helpers.py:
```diff
@@ -0,0 +1,241 @@
+    return None
+
+
+class TransactionType:
```
Hmmm this structure looks heavy. Why not to use an Enum? We already have one https://github.com/rotki/rotki/blob/d78b05575cd1299967db1bf7005a053778cf50f9/rotkehlchen/accounting/mixins/event.py#L14

nebolax commented on rotkehlchen/data_import/importers/_import_helpers.py:
```diff
@@ -0,0 +1,241 @@
+            base_asset: Asset,
+            timestamp: Timestamp,
+            link: str = '',
+        ) -> Trade | None:
```
We use `typing` module in our codebase, so I guess `Union[Optional[Trade]]` will be better

nebolax commented on rotkehlchen/data_import/importers/_import_helpers.py:
```diff
@@ -0,0 +1,241 @@
+            timestamp: Timestamp,
+            link: str = '',
+        ) -> Trade | None:
+            """
```
If a docstring fits in one line, we keep them on a single line

nebolax commented on rotkehlchen/data_import/importers/uphold_transactions.py:
```diff
@@ -63,138 +69,92 @@ def _consume_uphold_transaction(
         Information for the values that the columns can have has been obtained from sample CSVs
         """
-        with open(filepath, 'r', encoding='utf-8-sig') as csvfile:
+        with open(filepath, "r", encoding="utf-8-sig") as csvfile:
```
Why to change it? We tend to use single quotes

nebolax commented on rotkehlchen/data_import/importers/_import_helpers.py:
```diff
@@ -0,0 +1,241 @@
+from __future__ import annotations  # isort:skip
```
I don't think that we have files with names starting with an underscore

nebolax reviewed (commented):
@gahrb thank you for your work! I left some comments, but my general concern is that what is currently done may be not well integratable with other exchange importers.

Could you perhaps integrate other exchange importers with the "import helpers" that you are introducing in this PR? My concern is that each classification method (like `classify_transaction`, `is_trade`, etc.) will become smth like `if uphold: ... if binance: ... if cryptocom: ... if cointracking: ...` which does not sound good.

For quicker communication, you can join our discord https://discord.gg/fqafpQrk and chat with us in `dev-support` channel. Our discord server has a captcha verification, so have DMs open to be able to pass it.

LefterisJP commented:
This PR has stayed open for too long without any response from the author. Will close it soon if no progress is made.

LefterisJP commented:
Closing due to inactivity.
